src/jpeg.imageio/jpeginput.cpp

```diff
@@ -437,9 +437,9 @@ JpgInput::read_icc_profile(j_decompress_ptr cinfo, ImageSpec& spec)
     std::string errormsg;
     bool ok = decode_icc_profile(icc_buf, spec, errormsg);
     if (!ok) {
-        // errorfmt("Could not decode ICC profile: {}\n", errormsg);
-        // return false;
-        // Nah, just skip an ICC specific error?
+        errorfmt("Possible corrupt file, could not decode ICC profile: {}\n",
+                 errormsg);
+        return false;
```

Comment:
Initially I was a bit concerned that this would prevent loading of the jpeg itself as you alluded to, however, it looks like the reader here doesn't care about the return value of this function, so I think the image will still be read but would be up to the discretion of the original caller to see if they care about the error status being set or not... It's a bit risky as you say but I don't have a large corpus of jpegs around to know the extent of what might be lurking in terms of folks complaining about it.

Reply:
Yeah, I honestly don't know the right balance between trying to muddle through reading files that are partially broken, versus being more security-conservative and stopping as soon as something is clearly invalid in the file.

Perhaaps the right thing thing (in the long run, not in this PR) is to add a global attribute about whether/how much to be strict about abandoning reads of files that have anything invalid in them.